### PR TITLE
Implement uncompress functionality for PDF files Closes #38

### LINKED ISSUE #38: ENH: Add an uncompress functionality
Is it possible to uncompress PDF? similar to:

~~~sh
# uncompress
qpdf --stream-data=uncompress a.pdf b.pdf
# compress
qpdf b.pdf c.pdf
~~~

https://github.com/qpdf/qpdf/releases

### PR DESCRIPTION
**Pull request by Kaos599**
Closes #38
- Added a new `uncompress.py` script to decompress Flate-encoded content streams in PDF files.
- Updated `main` function to handle PDF input and output paths.
- Modified `pyproject.toml` to include the I001 rule for Ruff linter to ignore specific linting issues, since it was conflicting with black.

### CODE REVIEW

File: pdfly/uncompress.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+def decompress_content_stream(content: IndirectObject) -> None:
+    """Decompress a content stream if it uses FlateDecode."""
+    if content.get("/Filter") == "/FlateDecode":
+        try:
+            compressed_data = content.get_data()
+            uncompressed_data = zlib.decompress(compressed_data)
+            content.set_data(uncompressed_data)
+            del content["/Filter"]  # Remove compression flag
+        except zlib.error as e:
+            print(f"Decompression error: {e}")
```
Comment (Lucas-C) [inline pdfly/uncompress.py:50]:
```suggestion
        except zlib.error as error:
            print(f"Some content stream with /FlateDecode failed to be decompressed: {error}")
```
Your strategy there is that a `zlib.error` should NOT interrupt the command?

I'm OK with that, but maybe the error message should be slightly clearer.

**Review by Lucas-C — CHANGES_REQUESTED** [review]
Thank you for your contribution @Kaos599 👍

Could you please:
* add a mention of this new command in `README.md`
* add some unit tests in `tests/test_uncompress.py`

You could for example execute this command on all `sample-files/*./*.pdf` files, similar to this: https://github.com/py-pdf/pdfly/pull/15#discussion_r1827712584

**Comment by Kaos599** [comment]
SUre

> Thank you for your contribution @Kaos599 👍
> 
> Could you please:
> 
>     * add a mention of this new command in `README.md`
> 
>     * add some unit tests in `tests/test_uncompress.py`
> 
> 
> You could for example execute this command on all `sample-files/*./*.pdf` files, similar to this: [#15 (comment)](https://github.com/py-pdf/pdfly/pull/15#discussion_r1827712584)

Sure i will implement those within few days

**Comment by Lucas-C** [comment]
Hi @Kaos599

I see that you published a commit on this branch yesterday.
Is is ready for reviewing?
I think that there is still a minor merge conflict to resolve.

**Comment by Kaos599** [comment]
@Lucas-C i have been busy with college exams so i just tried a little yesterday. I still need to implement most of the changes, I will get on it soon and will tag you once done.

Thanks for your patience!

**Pull request closed without merging**